stdlib/src/collections/dict.mojo

```diff
@@ -702,6 +702,28 @@ struct Dict[K: KeyElement, V: CollectionElement](
             return default.value()[]
         raise "KeyError"
 
+    fn popitem(inout self) raises -> DictEntry[K, V]:
+        """Remove and return a (key, value) pair from the dictionary. Pairs are returned in LIFO order.
+        popitem() is useful to destructively iterate over a dictionary, as often used in set algorithms.
+        If the dictionary is empty, calling popitem() raises a KeyError.
+
+        Args: None
+
+        Returns:
+            Last dictionary item
+
+        Raises:
+            "KeyError" if the dictionary is empty.
+        """
+
+        for item in reversed(self.items()):
+            var key = item[].key
+            var value = item[].value
+            _ = self.pop(key)
```

Comment:
I'm afraid that in the future this will cause issue when the compiler is stricter, no? I would expect the compiler to prevent us from changing an object during iteration.

Reply:
I take a look what python does
```python
a = {'a': 1, 'b': 2}
for k, v in a.items():
    print(k, v)
    item = a.popitem()
    print(item)

```
Result
```
a 1
('b', 2)
Traceback (most recent call last):
  File "/Users/bytedance/mojo/sample.py", line 2, in <module>
    for k, v in a.items():
RuntimeError: dictionary changed size during iteration
```
I agree we may return error for modifying dict

Reply:
dang this is way harder than I imagined...